os/fs/smartfs/smartfs_smart.c

```diff
@@ -712,6 +712,72 @@ static int smartfs_sync_internal(struct smartfs_mountpt_s *fs, struct smartfs_of
 	return ret;
 }
 
+int allocate_new_sector(size_t buflen, struct smartfs_mountpt_s *fs, struct smartfs_chain_header_s *header, struct smart_read_write_s *readwrite, struct smartfs_ofile_s *sf)
```

Comment:
all function has 'smartfs' as a prefix, also only file struct's pointer is enough instead of all these params.